service/resource/endpoint/delete.go

```diff
@@ -43,6 +44,11 @@ func (r *Resource) ProcessDeleteState(ctx context.Context, obj, deleteState inte
 		return microerror.Mask(err)
 	}
 
+	if reflect.DeepEqual(endpointToApply, Endpoint{}) {
```

Comment:
This was necessary because `ProcessDelete` was called with `nil` if a pod with the annotation was updated. I am honestly unsure why this happens, but I added this because we do not care if the `deleteState` is empty.

Reply:
A refactor of this function is going to happen in #12 anyway. I just want to verify that everything works as intended first.